cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -2028,21 +2033,46 @@ pub(crate) fn define(
         e.enc_32_64(x86_psra, rec_fa.opcodes(*opcodes));
     }
 
-    // SIMD icmp using PCMPEQ*
-    for ty in ValueType::all_lane_types().filter(|t| t.is_int() && allowed_simd_type(t)) {
-        let (opcodes, isa_predicate): (&[_], _) = match ty.lane_bits() {
-            8 => (&PCMPEQB, None),
-            16 => (&PCMPEQW, None),
-            32 => (&PCMPEQD, None),
-            64 => (&PCMPEQQ, Some(use_sse41_simd)),
-            _ => panic!("invalid size for SIMD icmp"),
-        };
-
+    // SIMD integer comparisons
+    for (ty, cc, opcodes, isa_predicate) in &[
+        (I8, IntCC::Equal, &PCMPEQB[..], None),
+        (I16, IntCC::Equal, &PCMPEQW[..], None),
+        (I32, IntCC::Equal, &PCMPEQD[..], None),
+        (I64, IntCC::Equal, &PCMPEQQ[..], Some(use_sse41_simd)),
+        (I8, IntCC::SignedGreaterThan, &PCMPGTB[..], None),
+        (I16, IntCC::SignedGreaterThan, &PCMPGTW[..], None),
+        (I32, IntCC::SignedGreaterThan, &PCMPGTD[..], None),
+        (
+            I64,
+            IntCC::SignedGreaterThan,
+            &PCMPGTQ,
+            Some(use_sse42_simd),
+        ),
```

Comment:
It's a little unfortunate that the formatter is putting this one-liner over six lines.

Maybe some part of that could be factored out? `use INTCC::*`?